### ISSUE TITLE
feat(gauss_plume): likelihoods module with Student-t and heteroscedastic Gaussian noise

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#84]
## Problem / Request
`gauss_plume/inference.py::gaussian_plume_model` and the `gauss_puff` inference models each inline a Gaussian likelihood with a scalar noise std (#33 exposed `obs_noise_std` / `background_prior_std`). The roadmap's Tier I table lists `gauss_plume.likelihoods` ☐ and its open questions call for a Student-t (ν ≈ 5) default at low SNR because detection-floor effects make the noise heavy-tailed. Puff and plume should share the module, and the predictor (Step 5) needs the same likelihood for its training loss.

## User Story
> As a user fitting a faint plume, I want to switch the observation model to Student-t or a per-pixel heteroscedastic Gaussian without editing the NumPyro model, so the posterior isn't dominated by detection-floor outliers.

## Proposed API
```python
# src/plumax/gauss_plume/likelihoods.py
class Noise(Protocol):
    def __call__(self, name: str, mu: jax.Array, obs: jax.Array | None) -> jax.Array: ...  # numpyro.sample inside

@dataclass(frozen=True)
class GaussianNoise:        std: float | jax.Array
@dataclass(frozen=True)
class StudentTNoise:        scale: float | jax.Array; df: float = 5.0
@dataclass(frozen=True)
class HeteroscedasticNoise: std: jax.Array   # per-observation, e.g. from Instrument.R diagonal
```

## Implementation Steps
- [ ] Add `src/plumax/gauss_plume/likelihoods.py` with the three noise models (lazy NumPyro import, matching the PEP 562 pattern in `gauss_plume/__init__.py`)
- [ ] Add a `noise: Noise = GaussianNoise(...)` argument to `gaussian_plume_model` / `infer_emission_rate` and to the `gauss_puff.inference` models, defaulting to today's behaviour
- [ ] Document in the inference docstring the operational identifiability target from the design page: 3-overpass CV(Q) ≤ 50 %, single-overpass ≤ 100 %

## Definition of Done
- [ ] MAP recovery test with 5 % injected outliers: Student-t recovers Q within σ_post, Gaussian does not
- [ ] Existing `tests/gauss_plume/test_inference.py` and `tests/gauss_puff/` pass unchanged with the default
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/gauss_plume/test_likelihoods.py::test_student_t_robust_to_outliers` (`@pytest.mark.slow`)

## Relationships
- Parent (theme epic): #70
- Blocks: amortized NPE predictor (shares the loss)
- Related: #33 (closed; exposed the scalar noise std)
